website/compare.py

```diff
@@ -41,23 +34,36 @@ def load_metadata(file_path):
 
 st.title("Compare Solvers")
 
-# Set default solvers
-default_solver1 = "gurobi"
-default_solver2 = "highs"
-
-# Dropdown to select Solver 1 with default value Gurobi
+# Get list of unique solvers from the data
+solver_list = df["Solver"].unique()
+# Set default solvers based on available solver list length
+if len(solver_list) > 1:
+    default_solver1 = solver_list[0]
+    default_solver2 = solver_list[1]
+elif len(solver_list) == 1:
+    default_solver1 = solver_list[0]
+    default_solver2 = None
+else:
+    st.write("No solvers available for comparison.")
+    st.stop()
```

Comment:
```suggestion
else:
    st.write("Fewer than 2 solvers available for comparison. Please add more solvers to compare.")
    st.stop()
```
We can also simplify the code below to assume that there are at least 2 solvers.